examples/baseball.py

```diff
@@ -0,0 +1,273 @@
+ - Partial pooling model: A hierarchical model with partial data sharing.
+
+
+We recommend Radford Neal's tutorial on HMC ([3]) to users who would like to get a
+more comprehensive understanding of HMC and its variants, and to [4] for details on
+the No U-Turn Sampler, which provides an efficient and automated way (i.e. limited
+hyper-parameters) of running HMC on different problems.
+
+[1] Carpenter B. (2016), ["Hierarchical Partial Pooling for Repeated Binary Trials"]
+    (http://mc-stan.org/users/documentation/case-studies/pool-binary-trials.html).
+[2] Efron B., Morris C. (1975), "Data analysis using Stein's estimator and its
+    generalizations", J. Amer. Statist. Assoc., 70, 311-319.
+[3] Neal, R. (2012), "MCMC using Hamiltonian Dynamics",
+    (https://arxiv.org/pdf/1206.1901.pdf)
+[4] Hoffman, M. D. and Gelman, A. (2014), "The No-U-turn sampler: Adaptively setting
+    path lengths in Hamiltonian Monte Carlo", (https://arxiv.org/abs/1111.4246)
+"""
+
+logging.basicConfig(format='%(message)s', level=logging.INFO)
+# Enable validation checks
+pyro.enable_validation(True)
+pyro.set_rng_seed(1)
+DATA_URL = "http://www.swarthmore.edu/NatSci/peverso1/Sports%20Data/" + \
+           "JamesSteinData/Efron-Morris%20Baseball/EfronMorrisBB.txt"
+
+
+# ===================================
+#               MODELS
+# ===================================
+
+
+def fully_pooled(at_bats):
+    """
+    Number of hits in $K$ at bats for each player has a Binomial
+    distribution with a common probability of success, $\phi$.
+
+    :param (torch.Tensor) at_bats: Number of at bats for each player.
+    :return: Number of hits predicted by the model.
+    """
+    phi_prior = Uniform(at_bats.new_tensor(0), at_bats.new_tensor(1))
+    phi = pyro.sample("phi", phi_prior)
+    return pyro.sample("obs", Binomial(at_bats, phi))
+
+
+def not_pooled(at_bats):
+    """
+    Number of hits in $K$ at bats for each player has a Binomial
+    distribution with independent probability of success, $\phi_i$.
+
+    :param (torch.Tensor) at_bats: Number of at bats for each player.
+    :return: Number of hits predicted by the model.
+    """
+    num_players = at_bats.shape[0]
+    phi_prior = Uniform(at_bats.new_tensor(0), at_bats.new_tensor(1)).expand_by([num_players]).independent(1)
+    phi = pyro.sample("phi", phi_prior)
+    return pyro.sample("obs", Binomial(at_bats, phi))
+
+
+def partially_pooled(at_bats):
+    """
+    Number of hits has a Binomial distribution with a logit link function.
+    The logits $\alpha$ for each player is normally distributed with the
+    mean and scale parameters sharing a common prior.
+
+    :param (torch.Tensor) at_bats: Number of at bats for each player.
+    :return: Number of hits predicted by the model.
+    """
+    num_players = at_bats.shape[0]
+    loc = pyro.sample("loc", Normal(at_bats.new_tensor(-1), at_bats.new_tensor(1)))
+    scale = pyro.sample("scale", HalfCauchy(at_bats.new_tensor(0), at_bats.new_tensor(1)))
```

Comment:
why HalfCauchy instead of TruncNormal?

Reply:
@martinjankowiak  Although the Stan tutorial used HalfNormal,  I think that HalfNormal is not available yet in Pytorch and Pyro (not so sure). And I think that people use HalfCauchy more often.